libbpfgo.go

```diff
@@ -924,17 +956,21 @@ func (b *BPFMap) DeleteKey(key unsafe.Pointer) error {
 //
 // For example:
 //
-//  key := 1
-//  value := []byte{'a', 'b', 'c'}
-//  keyPtr := unsafe.Pointer(&key)
-//  valuePtr := unsafe.Pointer(&value[0])
-//  bpfmap.Update(keyPtr, valuePtr)
-//
+//	key := 1
+//	value := []byte{'a', 'b', 'c'}
+//	keyPtr := unsafe.Pointer(&key)
+//	valuePtr := unsafe.Pointer(&value[0])
+//	bpfmap.Update(keyPtr, valuePtr)
 func (b *BPFMap) Update(key, value unsafe.Pointer) error {
 	return b.UpdateValueFlags(key, value, MapFlagUpdateAny)
 }
 
 func (b *BPFMap) UpdateValueFlags(key, value unsafe.Pointer, flags MapFlag) error {
+	var val interface{} = (*BPFMap)(value)
+	m, ok := val.(*BPFMap)
+	if ok {
+		value = unsafe.Pointer(&m.fd)
+	}
```

Comment:
Should we handle the case when the type wasn't of `BPFMap`?

Reply:
In that case, we can just past the value as is. We only have a special case if the given value is a map.

I'm not sure if it's the best way to check a type from an unsafe pointer. If you know any better way let me know.

Reply:
I see, not really a thing we should be concerned about here. Anyhow, perhaps something like `(*(*BPFMap)(value))` would be clearer?

Reply:
It's still not an assertion though. It's casting. But I can still use it while assigning the interface. This could be a stupid effort, in any case I'll test if further. 

It sucks that the api doesn't accept bpf_map rather than a fd.

Reply:
This was a rather stupid attempt. Type casting panics if the underlying type doesn't match.